schema/derive/tests/ui_pass/derive_into_schema.rs

```diff
@@ -42,4 +42,14 @@ pub enum Enum {
     FortyTwo,
 }
 
+pub trait Trait {
+    type Assoc;
+}
+
+#[derive(IntoSchema)]
+#[schema(bounds = "T: Trait, T::Assoc: IntoSchema")]
```

Comment:
this API looks redundant

Reply:
It's required for instructions. Check out `iri.rs`

Reply:
I understand it's required, but why do we need to specify these bounds explicitly? At least the `T: Trait` can be inferred, if not both

Reply:
Well, it's more explicit as macro won't generate anything else, just user input. Also simplier to implement

Reply:
ok, stay lazy and make it a ticket

Reply:
I tried to implement auto-infering. A stuck at compiling stage and after some time got errors like this:

> error[E0275]: overflow evaluating the requirement `model::Value: iroha_schema::IntoSchema`
  --> data_model/src/account.rs:96:9
   |
96 |         IntoSchema,
   |         ^^^^^^^^^^
   |
   = note: required for `BTreeMap<name::model::Name, model::Value>` to implement `iroha_schema::IntoSchema`
   = note: 1 redundant requirement hidden
   = note: required for `metadata::model::Metadata` to implement `iroha_schema::IntoSchema`
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable
   = note: this error originates in the derive macro `IntoSchema` (in Nightly builds, run with -Z macro-backtrace for more info)

Looks endless loop of recursive bounds evaluation.

See also https://github.com/rust-lang/rust/issues/48214

Reply:
I think that's the reason why `derive_more::Display` doesn't do that.

My current implementation of `bounds` feels the same as `bounds` from `derive_more::Display`. Do you think it makes sense to infer `T: Trait` and mix that with user input?